module/VuFind/src/VuFind/Config/Feature/SettingPropertiesInterface.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+/**
+ * Interface for exposing setting properties in a configuration class.
+ *
+ * PHP version 8
+ *
+ * Copyright (C) The National Library of Finland 2025.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see
+ * <https://www.gnu.org/licenses/>.
+ *
+ * @category VuFind
+ * @package  Config
+ * @author   Aleksi Peebles <aleksi.peebles@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+
+namespace VuFind\Config\Feature;
+
+/**
+ * Interface for exposing setting properties in a configuration class.
+ *
+ * @category VuFind
+ * @package  Config
+ * @author   Aleksi Peebles <aleksi.peebles@helsinki.fi>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+interface SettingPropertiesInterface
```

Comment:
Just wondering if there is a better name for this -- at a glance it's not obvious what it means. The words "options," "settings" and "properties" can mean similar/overlapping things, and "setting" can be a verb or a noun. I haven't dug into this deeply to form an opinion of what a better name might be, but the current name immediately leaves me feeling a little confused. :-)

Reply:
I did wonder if this was a good name... I am happy to rename the classes to something you think would be better.

My thought process was: 
- A configuration file consists of settings.
- A setting consists of a key and value.
- A setting might have properties (or attributes, or something else?) such as:
  - The setting is required.
  - The setting can be localized (it can have different values for different locales).
  - The setting has only specific allowed values (this is not implemented yet in the new interface/trait). These could possibly be called options?